### PR TITLE
QA tests on Snap

### PR DESCRIPTION
**Pull request by keliramu**

### CODE REVIEW

File: magefiles/mage.go [hunk 1]
```diff
@@ -539,10 +539,22 @@ func (Test) Hardening(ctx context.Context) error {
...
 }
 
+// Run QA tests (arguments: {testGroup} {testPattern})
+func (Test) QA(ctx context.Context, testGroup, testPattern string) error {
+	mg.Deps(mg.F(buildPackageDocker, "deb", "-cover"))
+	return qa(ctx, testGroup, testPattern)
+}
+
+// Run QA tests (arguments: {testGroup} {testPattern})
+func (Test) QASnap(ctx context.Context, testGroup, testPattern string) error {
```
Comment (keliramu) [inline magefiles/mage.go:549]: Added `mage` target to execute QA tests on SNAP package. 
It should be executed under `systemd` environment - Snap package can be installed only in that environment. One choice is to use Ubuntu VM and execute tests there to not interfere work on host.
Did some adjustments in Python code - but not all tests are adopted.

File: test/qa/lib/daemon.py [hunk 2]
```diff
@@ -71,14 +76,18 @@ def uninstall_peer(ssh_client: ssh.Ssh):
...
+        #sh.sudo.snap("start", "nordvpn")
+        os.popen("sudo snap start nordvpn").read()
+    elif is_init_systemd():
+        #sh.sudo.systemctl.start.nordvpnd()
+        os.popen("sudo systemctl start nordvpn").read()
+    else:
+        # call to init.d returns before the daemon is actually started
+        _rewrite_log_path()
+        sh.sudo("/etc/init.d/nordvpn", "start")
+        while not is_running():
```
Comment (mariusSincovici) [inline test/qa/lib/daemon.py:89]: shouldn't while check for all types that the daemon started?

File: test/qa/lib/daemon.py [hunk 3]
```diff
@@ -91,13 +100,17 @@ def start_peer(ssh_client: ssh.Ssh):
...
+    if is_under_snap():
+        #sh.sudo.snap("stop", "nordvpn")
+        os.popen("sudo snap stop nordvpn").read()
+    elif is_init_systemd():
+        #sh.sudo.systemctl.stop.nordvpnd()
+        os.popen("sudo systemctl stop nordvpn").read()
+    else:
+        # call to init.d returns before the daemon is actually stopped
+        sh.sudo("/etc/init.d/nordvpn", "stop")
+        while is_running():
```
Comment (mariusSincovici) [inline test/qa/lib/daemon.py:112]: same here

File: test/qa/lib/daemon.py (hunk 2 above)
Comment (keliramu) [reply test/qa/lib/daemon.py:89]: Before my changes, `waiting` for daemon start was only for `init.d` - maybe there was a reason for that, so, I left the same way.

File: test/qa/lib/daemon.py (hunk 3 above)
Comment (keliramu) [reply test/qa/lib/daemon.py:112]: same as above

**Review by bartoszWojciechO — APPROVED** [review]

File: test/qa/lib/daemon.py (hunk 2 above)
Comment (mariusSincovici) [reply test/qa/lib/daemon.py:89]: Probably because it never ran on systemd env :smile:

**Review by mariusSincovici — APPROVED** [review]

**Pull request merged**